substrate/frame/bounties/src/benchmarking.rs

```diff
@@ -119,6 +119,16 @@ benchmarks_instance_pallet! {
 		Treasury::<T, I>::on_initialize(BlockNumberFor::<T>::zero());
 	}: _<T::RuntimeOrigin>(approve_origin, bounty_id, curator_lookup, fee)
 
+	approve_bounty_with_curator {
+		setup_pot_account::<T, I>();
+		let (caller, curator, fee, value, reason) = setup_bounty::<T, I>(0, T::MaximumReasonLength::get());
+		let curator_lookup = T::Lookup::unlookup(curator);
+		Bounties::<T, I>::propose_bounty(RawOrigin::Signed(caller).into(), value, reason)?;
+		let bounty_id = BountyCount::<T, I>::get() - 1;
+		let approve_origin = T::SpendOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
+		Treasury::<T, I>::on_initialize(BlockNumberFor::<T>::zero());
+	}: _<T::RuntimeOrigin>(approve_origin, bounty_id, curator_lookup, fee)
```

Comment:
we need to assert the event at least for this new bench to make sure your benchmarks does what is intended. you can check other pallets for examples. you can test it by running the tests with runtime-benchmarks feature enabled. no need to regenerate the weights.